tc/sgx/trusted_worker_manager/enclave/kme/workload_processor_kme.cpp

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file
+ * Implements base class WorkloadProcessorKME to create an
+ * Avalon KME workload processor.
+ */
+
+#include <string>
+#include "enclave_utils.h"
+#include "workload_processor_kme.h"
+
+std::map<std::string, WorkloadProcessorKME*> \
+    WorkloadProcessorKME::workload_processor_table;
+
+WorkloadProcessorKME::WorkloadProcessorKME() {
+    this->ext_work_order_info_kme = nullptr;
+}
+
+WorkloadProcessorKME::~WorkloadProcessorKME() {}
+
+WorkloadProcessorKME* WorkloadProcessorKME::RegisterWorkloadProcessorKME(
+    std::string workload_id, WorkloadProcessorKME* processor) {
+
+    Log(TCF_LOG_INFO, "Register KME Workload Processor - %s",
+       workload_id.c_str());
+    workload_processor_table[workload_id] = processor;
+    processor->ext_work_order_info_kme = new ExtWorkOrderInfoKME();
+    return processor;
+}  // WorkloadProcessorKME::RegisterWorkloadProcessorKME
+
+WorkloadProcessorKME* WorkloadProcessorKME::CreateWorkloadProcessor(
+    std::string workload_id) {
```

Comment:
I think we can reuse same function base `WorkloadProcessor` class. No need of this function. Return value of this function can be type cast to `WorkloadProcessorKME`

Reply:
Removed this function and calling base class method from work_order_processor_kme.cpp